core/src/kura.rs

```diff
@@ -915,21 +1023,40 @@ mod tests {
             type Dir = futures::stream::Iter<std::vec::IntoIter<io::Result<OsString>>>;
             type File = UnreliableFile;
 
-            async fn read_dir(&self, _path: PathBuf) -> io::Result<Self::Dir> {
+            async fn read_dir(&self, path: PathBuf) -> io::Result<Self::Dir> {
+                let mut path_str = path.to_str().ok_or(io::ErrorKind::InvalidInput)?;
+                let path_string;
+                if !path_str.ends_with("/") {
+                    path_string = format!("{}/", path_str);
+                    path_str = &path_string;
+                }
                 Ok(futures::stream::iter(
                     self.files
                         .keys()
                         .cloned()
-                        .map(Ok)
+                        .map(|f_os| {
+                            let f = f_os.to_str();
+                            if f.is_some() && f.unwrap().starts_with(path_str) {
+                                let path_buf : PathBuf = f.unwrap().strip_prefix(path_str).unwrap().into();
```

Comment:
[`strip_prefix()`](https://doc.rust-lang.org/std/path/struct.Path.html#method.strip_prefix) docs shows, that you don't need to add `/` sign before, it should work just fine without it. So you can remove `path_srt` manipulation above

Reply:
That's not why it's important. It's important that the / get's removed so that it doesn't get treated as a root directory. If you start with dir/dir1/file2 and remove dir you get /dir1/file2 which is not what we want.

Reply:
Yeah, that's what I'm talking about. Look at [this playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=930966cdc057634a1a1e8920161fb7d1)